### PR TITLE
Spell checks https links clean rebased(#9916)

### LINKED ISSUE #8314: UserGuide, change reference page for Seika Display Braille
in the  user Guide, the reference website for Seika display braille is seika-braille.com

But if you go there, you will find a german page with no capabilities to choose language and in the homepage there are no information about Seika Braille display.
I recomend to change it with the following page:

http://en.seika-braille.com/down/index.html

This is the download page where people can take the USB Driver.

### LINKED ISSUE #9508: English User Guide: Additions/Corrections
I've carried out an entire read through of the English User Guide, and noticed the following additions/corrections that I hope may be useful - original text will come first on each line, followed by a ">" symbol and my suggested addition/correction. It's worth noting that this is based on alpha-16944,00a63f0a so I totally understand that several of these may already be fixed.

NVDA is copyright 2006-2018 NVDA contributors. > NVDA is copyright 2006-2019 NVDA contributors.

If you have not yet got a copy of NVDA, you can download it from www.nvaccess.org > If you have not yet got a copy of NVDA, you can download it from https://www.nvaccess.org

If created, this shortcut will also be assigned a shortcut key of control+alt+n, allowing you to start NVDA at any time with this key stroke > If created, this shortcut will also be assigned a shortcut key of control+alt+n, allowing you to start NVDA at any time with this keystroke

Tapping twice will result in a double tap. Tapping 3 times will result in a triple tap > Tapping twice will result in a double-tap. Tapping 3 times will result in a triple-tap

it's possible to have gestures like a 2-finger triple tap, a 4-finger tap, etc > it's possible to have gestures like a 2-finger triple-tap, a 4-finger tap, etc

When you find the key you wish to press, double-tap the key or lift your finger, depending on options chosen from Touch Interaction Settings category of the NVDA Settings > When you find the key you wish to press, double-tap the key or lift your finger, depending on options chosen from the Touch Interaction Settings category of the NVDA Settings

To get to the NVDA menu from anywhere in Windows while NVDA is running, press NVDA+n on the keyboard or perform a 2-finger double tap on the touch screen > To get to the NVDA menu from anywhere in Windows while NVDA is running, press NVDA+n on the keyboard or perform a 2-finger double-tap on the touch screen

NVDA Menu
NVDA+n
NVDA+n
2-finger double tap > NVDA Menu
NVDA+n
NVDA+n
2-finger double-tap

Tells NVDA to pass the next key press straight through to the active application, even if it is normally treated as an NVDA key command > Tells NVDA to pass the next key press straight through to the active application - even if it is normally treated as an NVDA key command

sleep mode disables all NVDA commands and speech/braille output for the current application. This is most useful in applications that provide their own speech or screen reading features. Press this command again to disable sleep mode > sleep mode disables all NVDA commands and speech/braille output for the current application. This is most useful in applications that provide their own speech or screen reading features. Press this command again to disable sleep mode - note that NVDA will only retain Sleep Mode settings until it is restarted

Reports the Text in the clipboard if there is any > Reports the Text on the clipboard if there is any

Note: numpad keys require numlock key to be turned off to work properly > Note: numpad keys require the num lock to be turned off to work properly (two instances)

http://www.dessci.com/en/products/mathplayer/ > https://www.dessci.com/mathplayer

• Design Science MathType in Microsoft Word and PowerPoint. MathType needs to be installed in order for this to work. The trial version is sufficient > • Design Science MathType in Microsoft Word and PowerPoint. MathType needs to be installed in order for this to work. The trial version is sufficient and can be downloaded from https://www.dessci.com/en/products/mathtype/

These settings will be stored in the document as bookmarks compatible with other screen readers such as Jaws > These settings will be stored in the document as bookmarks compatible with other screen readers such as JAWS

These settings will be stored in the workbook as defined name ranges compatible with other screen readers such as Jaws > These settings will be stored in the workbook as defined name ranges compatible with other screen readers such as JAWS

For certain dialogs, you can press Apply button to let the settings take effect immediately without closing the dialog > For certain dialogs, you can press the Apply button to let the settings take effect immediately without closing the dialog

Just like input/output, if you are concerned about privacy, you should not set logging level to this option > Just like input/output, if you are concerned about privacy, you should not set your logging level to this option

If this is enabled, NVDA will automatically check for updated versions of NVDA and inform you when an update is available > If this is enabled, NVDA will automatically check for updated versions and inform you when an update is available

The speech category in the NVDA Settings dialog contains options that lets you change the speech synthesizer as well as voice characteristics for the chosen synthesizer > The speech category in the NVDA Settings dialog contains options that allow you to change the speech synthesizer as well as voice characteristics for the chosen synthesizer

If you are using the Espeak NG synthesizer that is packaged with NVDA, this is a combo box that lets you select the Variant the synthesizer should speak with. ESpeak NG's Variants are rather like voices, as they provide slightly different attributes to the eSpeak NG voice. Some variants will sound like a male, some like a female, and some even like a frog > If you are using the Espeak NG synthesizer that is packaged with NVDA, this is a combo box that lets you select the Variant the synthesizer should speak with. ESpeak NG's Variants are rather like voices, as they provide slightly different attributes to the eSpeak NG voice. Some variants will sound like a male, some like a female, and some even like a frog. If using a third-party synthesizer, you may also be able to change this value if your chosen voice supports it.

This option allows you to change the rate of your voice. This is a slider that goes from 0 to 100, (0 being the slowest, 100 being the fastest) > This option allows you to change the rate of your voice. This is a slider that goes from 0 to 100  - 0 being the slowest, 100 being the fastest

This option allows you to change the pitch of the current voice. It is a slider which goes from 0 to 100, (0 being the lowest pitch and 100 being the highest) > This option allows you to change the pitch of the current voice. It is a slider which goes from 0 to 100 - 0 being the lowest pitch and 100 being the highest.

This option is a slider which goes from 0 to 100, (0 being the lowest volume and 100 being the highest) > This option is a slider which goes from 0 to 100 - 0 being the lowest volume and 100 being the highest.

This option is a slider that lets you choose how much inflection (rise and fall in pitch) the synthesizer should use to speak with. (The only synthesizer that provides this option at the present time is eSpeak NG) > This option is a slider that lets you choose how much inflection (rise and fall in pitch) the synthesizer should use to speak with.

This checkbox allows you to toggle whether NVDA should switch speech synthesizer languages automatically if the text being read specifies its language. This option is enabled by default. Currently only the eSpeak NG synthesizer supports automatic language switching > This checkbox allows you to toggle whether NVDA should switch speech synthesizer languages automatically if the text being read specifies its language. This option is enabled by default.

When this checkbox is enabled, NVDA will include additional symbol pronunciation dictionaries when pronouncing characters and symbols > When this checkbox is checked, NVDA will include additional symbol pronunciation dictionaries when pronouncing characters and symbols

Usually, NVDA raises the pitch slightly for any capital letter, but some synthesizers may not support this well, so perhaps this option may be of use > Usually, NVDA raises the pitch slightly for any capital letter, but some synthesizers may not support this well, so this option may be a preferred alternative for some users.

This option allows you to choose the sound card that NVDA should instruct the selected synthesizer to speak through > This option allows you to choose the audio device that NVDA should instruct the selected synthesizer to speak through

The Keyboard category in the NVDA Settings dialog contains options that sets how NVDA behaves as you use and type on your keyboard > The Keyboard category in the NVDA Settings dialog contains options that set how NVDA behaves as you use and type on your keyboard

The first checkbox lets you control if NVDA should use the capslock as an NVDA modifier key > The first checkbox lets you control if NVDA should use the caps lock as an NVDA modifier key

NVDA can be configured so that the numpad Insert, Extended Insert and/or capslock key can be used > NVDA can be configured so that the numpad Insert, Extended Insert and/or caps lock key can be used

e.g. you wish to turn capslock on when you have set capslock to be an NVDA modifier key > e.g. you wish to turn caps lock on when you have set caps lock to be an NVDA modifier key

The Desktop layout makes heavy use of the numpad (with numlock off) > The Desktop layout makes heavy use of the numpad (with num lock off)

If your laptop cannot do this or does not allow you to turn numlock off > If your laptop cannot do this or does not allow you to turn num lock off

If this is unchecked, you need to double-tap on the touch keyboard key to press the key > If this is unchecked, you need to double-tap on the key on the touch keyboard to press the key

Report Tool Tips > Report Tooltips

A checkbox that when checked tells NVDA to report tool tips as they appear. Many Windows and controls show a small message (or tool tip > A checkbox that when checked tells NVDA to report tooltips as they appear. Many Windows and controls show a small message (or tooltip

Help Balloons are like tool tips > Help Balloons are like tooltips

Uncheck this checkbox if you don't wish to have the description reported along with objects - it may be useful to give an example of a typical description that NVDA may encounter here

For example, when you enter text into the search box in Start menu in Windows Vista and later > For example, when you enter text into the search box in the Start menu in Windows 8 and later

As an example, if enabled, if the letter j was pressed, it would be trapped from reaching the document, even though it is not a quick navigation command nor is it likely to be a command in the application itself > As an example, if enabled, if the letter j was pressed, it would be trapped from reaching the document, even though it is not a quick navigation command nor is it likely to be a command in the application itself. If enabled, NVDA will tell Windows to play a default sound whenever a key is pressed that gets trapped.

When this option is enabled, NVDA will try to use the Microsoft UI Automation accessibility api in order to fetch information from Microsoft Word document controls > When this option is enabled, NVDA will try to use the Microsoft UI Automation accessibility API in order to fetch information from Microsoft Word document controls

Section 11.3 - we may wish to remove the laptop key assignments; they match their desktop counterparts

If the add-on was previously "enabled", a status will show "disabled after restart" Just like when you install or remove add-ons, you need to restart NVDA in order for changes to take effect > If the add-on was previously "enabled", the status will show "disabled after restart" - just like when you install or remove add-ons, you need to restart NVDA in order for changes to take effect

The Incompatible Add-ons Manager, which can be accessed via the "view incompatible add-ons" buttons in the Add-on manager > The Incompatible Add-ons Manager, which can be accessed via the "view incompatible add-ons" button in the Add-on manager

The Incompatible add-ons manager also has an "About add-on..." button. This opens will let you know the full details of the add-on, which is helpful when contacting the add-on author > The Incompatible add-ons manager also has an "About add-on..." button. This screen will let you know the full details of the add-on, which is helpful when contacting the add-on author

Please see extra voices page > please see the extra voices page

http://www.microsoft.com/download/en/details.aspx?id=27225 > https://www.microsoft.com/download/en/details.aspx?id=27225

http://www.microsoft.com/download/en/details.aspx?id=27224 > https://www.microsoft.com/download/en/details.aspx?id=27224

Quit already running copy of NVDA > Quit an already running copy of NVDA

Quit already running copy of NVDA and start this one > Quit an already running copy of NVDA and start this one

http://www.nvaccess.org/ > https://www.nvaccess.org/

### PR DESCRIPTION
**Pull request by Adriani90**
### Link to issue number:
fixes #9508 
fixes #8314

### Summary of the issue:
* There are some examples missing in the user guide (i.e. regarding object description)and some text passages should be deleted (i.e. that eSpeak is the only synthesizer which suports infelction and change in pitch or automatic language switch). Also there are some miss spellings. Most links are still http: and not https:
* Some links for braille displays are outdated (i.e. baum and Seika).

### Description of how this pull request fixes the issue:
• Edited the user guide accordingly and fixed the spelling errors. In the buildVersionInfo, the URL is a https: now as well as all other links in the user guide.
• Links for Seika and Baum braille displays are up to date.

### Testing performed:
Ran NVDA from source and checked that formating is not changed and that links are working. Also tested that the message in global commands is reported properly by NVDA.

### Known issues with pull request:
none

### Change log entry:
none

### CODE REVIEW

**Comment by Adriani90** [comment]
Uh that was crazy :)) now it seems it passed all checks. @feerrenrut sorry for the confusions, but now I think I understand where my submodule changes came from. I think I forgot to run git submodule update after pulling. Well, now I think it should be ok.

File: source/globalCommands.py [hunk 1]
```diff
@@ -1326,7 +1326,8 @@ def script_showGui(self,gesture):
 	def script_review_sayAll(self,gesture):
 		sayAllHandler.readText(sayAllHandler.CURSOR_REVIEW)
 	# Translators: Input help mode message for say all in review cursor command.
-	script_review_sayAll.__doc__ = _("Reads from the review cursor up to end of current text, moving the review cursor as it goes")
+	script_review_sayAll.__doc__ = _("Reads from the review cursor up to the end of the current text, \
```
Comment (feerrenrut) [inline source/globalCommands.py]: Rather than using '\' to split these lines, you can just have a string on each line. Strings are automatically joined in line continuations like this. The brackets groups them and means they will be joined together. Though note the space at the star of the first string.

``` python
	script_review_sayAll.__doc__ = _(
		"Reads from the review cursor up to the end of the current text,"
		" moving the review cursor as it goes"
	)
```

Also note that the first string starts on a new line after the opening bracket, this is "hanging indent" style, and will keep the lint checks happy

File: user_docs/en/userGuide.t2t [hunk 2]
```diff
@@ -592,12 +592,12 @@ A key command is provided to return to the original page containing the embedded
...
 Using MathPlayer 4 from Design Science, NVDA can read and interactively navigate supported mathematical content.
 This requires that MathPlayer 4 is installed on the computer.
-MathPlayer is available as a free download from: http://www.dessci.com/en/products/mathplayer/
+MathPlayer is available as a free download from: https://www.dessci.com/en/products/mathplayer/
 
 NVDA supports the following types of mathematical content:
 - MathML in Mozilla Firefox, Microsoft Internet Explorer and Google Chrome.
 - Design Science MathType in Microsoft Word and PowerPoint.
-MathType needs to be installed in order for this to work. The trial version is sufficient.
+MathType needs to be installed in order for this to work. The trial version is sufficient. It can be downloaded from https://www.dessci.com/en/products/mathtype/
```
Comment (feerrenrut) [inline user_docs/en/userGuide.t2t]: Can you put this new sentence on a new line please. Single line breaks are removed when this is rendered. We separate sentences to minimize translations. If you put this new sentence on a new line, then it will only be the new sentence and not three that require re-translation.

File: user_docs/en/userGuide.t2t [hunk 3]
```diff
@@ -1009,8 +1009,8 @@ The available logging levels are:
 - Disabled: Apart from a brief startup message, NVDA will not log anything while it runs.
 - Info: NVDA will log basic information such as startup messages and information useful for developers.
 - Debug warning: Warning messages that are not caused by severe errors will be logged.
-- Input/output: Input from keyboard and braille displays, as well as speech and braille output will be logged. If you are concerned about privacy, do not set logging level to this option.
-- Debug: In addition to info, warning, and input/output messages, additional debug messages will be logged. Just like input/output, if you are concerned about privacy, you should not set logging level to this option.
+- Input/output: Input from keyboard and braille displays, as well as speech and braille output will be logged. If you are concerned about privacy, do not set the logging level to this option.
```
Comment (feerrenrut) [inline user_docs/en/userGuide.t2t]: Again for each of these, please split sentences onto new lines. You should also double check that this gets rendered correctly as part of the list.

File: user_docs/en/userGuide.t2t [hunk 4]
```diff
@@ -1485,7 +1485,7 @@ It has the following options:
 -
 
 ==== Report background progress bars ====[ObjectPresentationReportBackgroundProgressBars]
-This is an option that, when checked, tells NVDA to keep reporting a progress bar, even if it is not physically in the foreground.
+This is an option that, when checked, tells NVDA to keep reporting a progress bar - even if it is not physically in the foreground.
```
Comment (feerrenrut) [inline user_docs/en/userGuide.t2t]: I think this should still be a comma.

File: user_docs/en/userGuide.t2t [hunk 5]
```diff
@@ -1497,7 +1497,7 @@ Toggles the announcement of new content in particular objects such as terminals
 ==== Play a sound when auto-suggestions appear ====[ObjectPresentationSuggestionSounds]
 Toggles announcement of appearance of auto-suggestions, and if enabled, NVDA will play a sound to indicate this.
 Auto-suggestions are lists of suggested entries based on text entered into certain edit fields and documents.
-For example, when you enter text into the search box in Start menu in Windows Vista and later, Windows displays a list of suggestions based on what you typed.
+For example, when you enter text into the search box in Start menu in Windows 8 and later, Windows displays a list of suggestions based on what you typed.
```
Comment (feerrenrut) [inline user_docs/en/userGuide.t2t]: Why is this changed? Have you tested that this? There is a search box in the start menu of vista, and windows 7 which gives suggestions, do these not work with NVDA?

**Review by feerrenrut — CHANGES_REQUESTED** [review]

File: source/globalCommands.py (hunk 1 above)
Comment (Adriani90) [reply source/globalCommands.py]: Thanks very much, this is a good idea.
> Though note the space at the star of the first string.
You mean the last string?

One more question. If I split the line after the bracket, will this not cause errors in the gettext function for translators? After some research my understanding was that I could split like this, including the _ in the string:

	script_review_sayAll.__doc__ = (
		"_(Reads from the review cursor up to the end of the current text,"
		" moving the review cursor as it goes")
	)

But if this does not have an impact, then I would prefer your aproach.

File: user_docs/en/userGuide.t2t (hunk 5 above)
Comment (Adriani90) [reply user_docs/en/userGuide.t2t]: Ok I was prety sure that Windows introduced autocomplete feature to start menu in Windows 8 and I thought in Windows 7 and Vista this feature was present only in Windows explorer and the run box. Testing shows that suggestions are also shown in Windows 7. I don't know how it is in Windows Vista though. But I changed it back to Windows Vista. I think most users will not be confused by this.

Comment (lukaszgo1) [reply user_docs/en/userGuide.t2t]: The suggestions are  also available on Windows  Vista. Having said that in my opinion it should be reworded. For a beginnner user reading this the immediate conclusion would be that if he/she is using Windows Vista or later when typing in the start menu the suggestion sound would play when suggestion appears. It is not the case ad least not on anything lover than Windows 10. To avoid this confusion it might make sense either to not use an example here, or note that it works only in Windows 10.

**Comment by Adriani90** [comment]
Hmm could we not just say that this works only in Windows 10 (i.e. when typing something in the search field in the start menu? @eliot94 does the auto suggestion sound work for you in Windows 8? I ask you because you proposed that wording.

**Comment by elliott94** [comment]
I'm not sure if these sounds are played on Windows 8; my assumption at the time was that they would be due to the changes introduced with the OS, but I never confirmed this. Sorry. :(

Technically, could we introduce this behaviour for Windows Vista and later? I realise that this is beyond the scope of this particular pull, but happy to log if somebody thinks that this would physically be possible to implement.

**Comment by lukaszgo1** [comment]
@elliott94 wrote;

> 
> 
> I'm not sure if these sounds are played on Windows 8; my assumption at the time was that they would be due to the changes introduced with the OS, but I never confirmed this. Sorry. :(

I've tested on Windows 8.1 and they aren't.

> 
> Technically, could we introduce this behaviour for Windows Vista and later? I realise that this is beyond the scope of this particular pull, but happy to log if somebody thinks that this would physically be possible to implement.


Well this is open source. No one can stop you from writing anything :-).

**Comment by Adriani90** [comment]
Ok, then I will change the example for autosuggestion sounds to the Windows 10 start menu. @josephsl do I understand your comment in issue #8912 correctly, that app modules and addons can request playing sounds for autosuggestions in windows 8 and windows 7?

**Comment by josephsl** [comment]
Hi,

Auto-complete and auto-suggest are two completely different things, although they share the underlying theme:

* Auto-complete: the current line will be filled with most likely result automatically.
* Auto-suggest: a list of possible suggestions will appear.

Proper support for auto-suggest accessibility is included (at least in a minimal form) in Windows 8.x. Windows 10 uses this in a number of places in addition to Start menu. Because of limited scope of auto-suggest accessibility in Windows 8.x, I skipped directly to Windows 10 (after all, the true source of my code in 2017 was Windows 10 App Essentials).

The code I wrote in 2017 is abstract enough that it can be used in places where the control can communicate appearance of suggestions - in fact, @DerekRiemer took advantage of this in his Notepad++ add-on. In order for it to work properly, the control the NVDA is interested in must communicate to NVDA that suggestions have appeared. This works beautifully in Windows 10 because this is taken care of by UIA controller for events coming from search edit fields. The UIA controller for event handler for search field class, which is powered by a behavior class named "editable text with suggestions", tells the behavior class to play appropriate sound when suggestions appear and disappear.

Fpr things other than UIA search fields in Windows 10 and certain apps, cooperation between NVDA, the control in question, and suggestions list control must be used: NVDA would monitor appearance of suggestions, and this in turn is communicated to users via sound or other means. This is how Notepad++ add-on can notify you of code suggestions. Because suggestion sound playback is handled via a behavior class, it can be extended by overlay classes, and I designed it that way with several scenarios in mind, most notably IntelliSense suggestions in Visual Studio.

In the course of two years or more, I and others discovered oddities here and there. For example, not all search fields report top suggestion automatically - search box in Settings app doesn't. Some search fields do not even raise UIA controller for event yet it should be treated as a search field - contacts search box in People app is a notable case. Some search fields are no longer treated as such because it causes problems - classic Edge's address omnibar is a notable one in that NVDA played suggestion sound whenever Edge's window was maximized, caused by the address omnibar firing UIA controller for event.

It might be possible to extend this to Windows 8.x (perhaps not Windows 7 unless something can be arranged; at this point I'm personally against that as Windows 7 is leaving this world soon as far as support from Microsoft is concerned). The problem is dealing with events - if suggestions in Start screen is proceeded by a single event (say, window opened event), then it might be possible to catch it and do something; however, if many events are fired (say, element selected event whenever search suggestions change), then it may lead to frustration from users.

As someone suggested, I think discussion of auto-suggest announcement feature for Windows 7 and 8.x deserves a dedicated issue. Therefore, I advise against editing the user guide by saying that suggestion sound is available on earlier Windows releases - leave it as is.

Thanks.

File: source/globalCommands.py (hunk 1 above)
Comment (Adriani90) [reply source/globalCommands.py]: @feerrenrut I think I have addressed all the review actions here. Could you please check the https://github.com/nvaccess/nvda/pull/10130#discussion_r317332023 ? Especially the hanging indent in the input message? I am not sure if the implementation as it is now in this PR would affect the translation.

**Comment by Adriani90** [comment]
But I see there are conflicts here so actually it might not be possible to merge. I have to create a new one I guess.

**Comment by lukaszgo1** [comment]
@Adriani90 Why you want to close this one rather than fix merge conflicts?

**Comment by Adriani90** [comment]
@lukaszgo1 unfortunately I don't really know how to solve the conflicts. I still have to learn this. If you can give me some help with that I certainly prefer to solve them instead of creating a new PR.

**Comment by feerrenrut** [comment]
I can resolve the conflicts for you, or if you would prefer to try yourself, I would suggest reading [githubs guide](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/resolving-a-merge-conflict-using-the-command-line). I have some slightly shorter instructions, specific to your case:

After doing `git fetch`, do `git merge origin/master` which will show that there are conflicts:
```
$ git merge origin/master
Auto-merging user_docs/en/userGuide.t2t
CONFLICT (content): Merge conflict in user_docs/en/userGuide.t2t
Auto-merging source/globalCommands.py
Automatic merge failed; fix conflicts and then commit the result.
```
This tells you where there has been changes both in your branch and in the branch you are merging in (origin/master). Git is unable to determine what the correct result should be.

Resolve the conflicts:
1. Open the 'CONFLICT' file with a text editor eg Notepad++
1. Search for the conflict marker: `<<<<<<<`. Notice that after this it has "HEAD", the following line is the line as it appears in the commit pointed to by "HEAD" which, is essentially, your branch. There is then a line `=======` which is a separator for the conflicting line that follows. After the conflicting line there is `>>>>>>> origin/master` which tells you that the previous line (the conflicting line) is from `origin/master`
1. You may want to go back to the command line and look at the history of your branch, and the history of master to understand each of these changes.
1. Modify one of the lines as necessary, and delete the other and delete the conflict marker lines (starting with: `<<<<<<<`, `=======`, `>>>>>>>`). Normally you shouldn't make other modifications to the file, if these are necessary, do them in a follow up commit. 
1. Search again for conflict markers to ensure there are no other conflicts.
1. Save the file. 
1. Stage the file with `git add -- user_docs/en/userGuide.t2t`
1. Commit `git commit`

File: source/globalCommands.py (hunk 1 above)
Comment (feerrenrut) [reply source/globalCommands.py]: I believe that the `checkPot` test (which runs as part of a PR build) should catch any errors here. If you are uncertain, check for the string in the created `.pot` file (after merging master, it will be in the artifacts tab for your PR build). Putting the `_(` inside the string will not work, since the underscore or "line" is actually just a short name for a function that looks up the translation.

**Comment by Adriani90** [comment]
@feerrenrut thanks very much for the guidance. It helped alot. However, it seems during the merge it also added the commits before july 7th 2019 which were from the old PR. I think it was a cherry picking effect because I cherry picked part of the commits from there.
So you reviewed the commits which are after July 7th. How can I exclude the commits before July 7th from this PR? I don't want to have them here because actually I want to open new PRs for them.

**Comment by feerrenrut** [comment]
If the changes you want to discard are contained within commits, you could do an interactive rebase (`git rebase -i origin/master`) this will open your configured text editor automatically (the same one that opens when you do git commit) and you be presented with essentially three columns: the action ('pick' is the default), the commit SHA, then the first line of the commit message. The rows are in the order they will apply.

You can then change the word 'pick' at the start of the line to 'd' or 'drop' on the lines of the commits you wish to discard.

You might wish to first create a new branch so that you can easily reset in case you make a mistake.
So you might do:
`git branch backupMyCurrentBranch`
`git rebase -i origin/master`

Notice you made a mistake:
`git reset --hard backupMyCurrentBranch`
Then try the rebase again.

**Comment by feerrenrut** [comment]
Ah, I can see what has happened. The master branch on your NVDA repo (Adriani90/nvda) has extra commits, ahead of nvaccess/nvda. It looks like you did two merges 16 hours ago, one of them I think was from your repo's master branch. 

Instead of a rebase, and dropping commits, you might want to consider just doing the merge again. The commit before the merge was `41c94e3` so to go back to there: `git reset --hard 41c94e3`.
Restart the merge `git merge nvaccess/master`
Resolve conflicts, and then you will have to force push your branch.

**Comment by feerrenrut** [comment]
As an aside, it is probably a good idea to do a hard reset of your master branch and force push it if you do not want those commits .

**Comment by Adriani90** [comment]
Hmm strange, now I have managed to exclude all unwanted commits from this pr but it seems there is a file changed (comtypes) that I didn't change. All the commits as of now are reviewed by you @feerrenrut. Is it possible to merge this PR without considering that comtypes file? I might come because I ran submodule update at some point.

**Comment by Adriani90** [comment]
@feerrenrut yes I will definitely hard reset my master branch after this :)) I have the other commits backed up on another branch anyway.

**Comment by Adriani90** [comment]
Huh ok a new rebase solved this actually. Now everything should be fine. :-)

**Review by feerrenrut — APPROVED** [review]
Thanks @Adriani90
